test/pls.jl

```diff
@@ -673,6 +673,39 @@ end
+        lme4_cooks = [0.1270714, 0.1267805, 0.243096, 0.0002437091, 0.03145029, 0.2954052, 0.04550505,
+                      0.3552723, 0.1984806, 0.4518805, 0.1683441, 0.02902698, 0.004232616, 1.734029e-05,
+                      0.003816645, 0.00623334, 0.03219321, 0.05429389, 0.07319191, 0.06649928, 0.007803994,
+                      0.001435875, 0.03886176, 0.01013682, 7.076106e-05, 0.02487801, 0.01538649, 0.002299068,
+                      0.008366248, 0.08733211, 0.3043884, 0.0770035, 0.003193764, 0.000259058, 0.00841487,
+                      0.00664586, 0.0894498, 0.007342141, 0.07721502, 0.00115366, 0.0476889, 0.01107893,
+                      0.02342937, 0.04474152, 0.009826393, 0.02536012, 0.07157197, 8.781548e-08, 0.1757661,
+                      0.01755979, 0.04308501, 0.04907289, 0.003603381, 0.02141832, 0.01529109, 0.0002237688,
+                      1.055383, 0.01226195, 0.01122611, 0.7032865, 0.01801972, 0.008351314, 0.009071886,
+                      1.922539e-05, 0.009401271, 0.01932602, 0.0001153177, 0.003751265, 0.02194446, 4.78793e-09,
+                      0.02048001, 0.01981013, 0.04247507, 0.03844668, 0.007580713, 0.01639404, 0.001973649,
+                      0.006080187, 0.0008513994, 0.08466273, 0.0878464, 0.2161317, 0.0467594, 0.06665132,
+                      0.0006486227, 0.0009503809, 0.03397066, 0.1231246, 0.1946271, 0.2816787, 0.008455713,
+                      0.02639438, 0.1743106, 0.00450064, 1.73262e-05, 0.01563701, 0.01998501, 0.02539804,
+                      0.157366, 0.1206117, 0.002382807, 0.007197368, 0.009506474, 0.002782844, 0.02747835,
+                      0.00986326, 0.008074464, 0.001298994, 0.03273043, 0.05191876, 0.005918988, 0.0696993,
+                      0.05733613, 0.1038886, 0.0881868, 0.008494316, 0.159206, 0.03677518, 0.135499, 0.06079108,
+                      0.003406159, 0.1399327, 0.001825492, 0.00191708, 0.01107303, 0.004549203, 0.02109569,
+                      0.1587737, 0.002198379, 0.006746796, 0.3064917, 3.780973e-07, 0.02104387, 0.04698987,
+                      0.02207251, 0.009852787, 0.0009590272, 1.506034e-05, 0.001194266, 0.003147009, 0.01284797,
+                      1.315739e-05, 0.03073671, 0.00899036, 0.01262709, 0.002494427, 0.03239389, 0.01698841,
+                      0.0002320865, 0.0135889, 0.02761053, 0.02916589, 0.04618232, 0.07875934, 0.02248172,
+                      0.1308213, 0.04340534, 0.05379937, 0.0873526, 0.07648689, 0.03333461, 0.01267992,
+                      0.004915966, 0.0003118122, 0.006997041, 0.01519545, 0.162238, 0.01767151, 0.02365221,
+                      0.05187042, 1.31043e-07, 0.002747362, 0.003266733, 0.005808394, 0.03485179, 0.003650455,
+                      0.0003004733, 1.535027e-05, 0.0168071, 1.510735e-05]
+        model = refit!(first(models(:sleepstudy)); progress=false)
+        @test all(zip(lme4_cooks, cooksdistance(model))) do (x, y)
+            return isapprox(x, y; atol=1e-5)
+        end
```

Comment:
For my own edification, is there a reason why you want to check component-wise approximate equality here rather than approximate equality of the arrays? I don't have a good intuition for when one approach is preferable to the other.

Reply:
Component-wise is stronger (which isn't always a good thing). I wanted it here because Cook's distance is used to evaluate individual observations, so we want to make sure it aligns at the level of individual observations.

On the other hand, if you were thinking about whether two points in some higher dimensional space are close, then you would care whether the vectors representing them are approximately equal not whether each coordinate is approximately equal. (Although this works out to the same thing in some norms, but not in L2.)

Reply:
Makes perfect sense. Thanks, I appreciate the explanation!